src/generic_core/core.ts

```diff
@@ -439,18 +439,35 @@ class uProxyCore implements uProxy.CoreAPI {
 
   public sendFeedback = (feedback :uProxy.UserFeedback) : void => {
     var xhr = freedom["core.xhr"]();
-    var postRequest =
-      'https://www.uproxy.org/submit-feedback?'
-        + 'email=' + encodeURIComponent(feedback.email) + '&'
-        + 'feedback=' + encodeURIComponent(feedback.feedback) + '&'
-        + 'logs=' + encodeURIComponent('logsPlaceholder');
-    xhr.open('POST', postRequest, true);
-    xhr.send();
+    if (feedback.logs) {
+      this.getLogs().then((formattedLogs) => {
+        var params = JSON.stringify(
+          {'email' : feedback.email,
+           'feedback' : feedback.feedback,
+            'logs' : formattedLogs});
+        xhr.open('POST', 'https://www.uproxy.org/submit-feedback', true);
+        xhr.send({'string': params});
+      });
+    } else {
+      var params = JSON.stringify(
+        {'email' : feedback.email,
+         'feedback' : feedback.feedback,
+          'logs' : ''});
+      xhr.open('POST', 'https://www.uproxy.org/submit-feedback', true);
+      // core.xhr requires the parameters to be tagged as either a
+      // string or array buffer in the format below.
+      // This is roughly equivalent to standard xhr.send(params).
+      xhr.send({'string': params});
```

Comment:
This and the above case are almost identical except in the second one logs is set to ''.  Can this be refactored into a single function?  You could make the function scoped within sendFeedback